vector/src/main/java/im/vector/app/features/settings/devices/v2/ParseDeviceUserAgentUseCase.kt

```diff
@@ -0,0 +1,148 @@
+        } else {
+            deviceModel = deviceInfoSegments.getOrNull(0)
+            deviceOperatingSystem = deviceInfoSegments.getOrNull(1)
+        }
+        return DeviceUserAgent(DeviceType.MOBILE, deviceModel, deviceOperatingSystem, appName, appVersion)
+    }
+
+    private fun parseIosUserAgent(userAgent: String): DeviceUserAgent {
+        val appName = userAgent.substringBefore("/")
+        val appVersion = userAgent.substringAfter("/").substringBefore(" (")
+        val deviceInfoSegments = userAgent.substringAfter("(").substringBeforeLast(")").split("; ")
+        val deviceModel = deviceInfoSegments.getOrNull(0)
+        val deviceOperatingSystem = deviceInfoSegments.getOrNull(1)
+        return DeviceUserAgent(DeviceType.MOBILE, deviceModel, deviceOperatingSystem, appName, appVersion)
+    }
+
+    private fun parseDesktopUserAgent(userAgent: String): DeviceUserAgent {
+        val browserSegments = userAgent.split(" ")
+        val browserName = when {
+            isFirefox(browserSegments) -> {
+                "Firefox"
+            }
+            isEdge(browserSegments) -> {
+                "Edge"
+            }
+            isMobile(browserSegments) -> {
+                getMobileBrowserName(browserSegments)
+            }
+            isSafari(browserSegments) -> {
+                "Safari"
+            }
+            else -> {
+                getRegularBrowserName(browserSegments)
+            }
+        }
+
+        val deviceOperatingSystemSegments = userAgent.substringAfter("(").substringBefore(")").split("; ")
+        val deviceOperatingSystem = if (deviceOperatingSystemSegments.getOrNull(1)?.startsWith("Android").orFalse()) {
+            deviceOperatingSystemSegments.getOrNull(1)
+        } else {
+            deviceOperatingSystemSegments.getOrNull(0)
+        }
+        return DeviceUserAgent(DeviceType.DESKTOP, browserName, deviceOperatingSystem, null, null)
+    }
+
+    private fun parseWebUserAgent(userAgent: String): DeviceUserAgent {
+        return parseDesktopUserAgent(userAgent).copy(
+                deviceType = DeviceType.WEB
+        )
+    }
+
+    private fun createUnknownUserAgent(): DeviceUserAgent {
+        return DeviceUserAgent(DeviceType.UNKNOWN)
+    }
+
+    private fun isFirefox(browserSegments: List<String>): Boolean {
+        return browserSegments.lastOrNull()?.startsWith("Firefox").orFalse()
+    }
+
+    private fun isEdge(browserSegments: List<String>): Boolean {
+        return browserSegments.lastOrNull()?.startsWith("Edge").orFalse()
+    }
+
+    private fun isSafari(browserSegments: List<String>): Boolean {
+        return browserSegments.lastOrNull()?.startsWith("Safari").orFalse() &&
+                browserSegments.getOrNull(browserSegments.size - 2)?.startsWith("Version").orFalse()
+    }
+
+    private fun isMobile(browserSegments: List<String>): Boolean {
+        return browserSegments.lastOrNull()?.startsWith("Safari").orFalse() &&
```

Comment:
Out of curiosity, why do we check for `Safari` keyword for mobile browser case?

Reply:
Nice catch! Fixed.